src/t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.hxx

```diff
@@ -0,0 +1,185 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2023 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_geometry_lagrange.hxx
+ * This geometry implements curved geometries obtained by Lagrange interpolation.
+ * The interpolation is carried out with the classical finite element basis functions.
+ * Therefore, it is the generalization of the linear geometry (t8_geometry_linear).
+ */
+
+#ifndef T8_GEOMETRY_LAGRANGE_HXX
+#define T8_GEOMETRY_LAGRANGE_HXX
+
+#include <string>
+#include <vector>
+
+#include <t8.h>
+#include <t8_cmesh/t8_cmesh_types.h>
+#include <t8_geometry/t8_geometry_with_vertices.hxx>
+#include <t8_geometry/t8_geometry_with_vertices.h>
+
+/* Mapping with Lagrange basis functions */
+struct t8_geometry_lagrange: public t8_geometry_with_vertices
+{
+ public:
+  /** 
+   * Constructor of the Lagrange geometry with a given dimension. The geometry
+   * is compatible with all tree types and uses as many vertices as the number of Lagrange
+   * basis functions used for the mapping.
+   * The vertices are saved via the \ref t8_cmesh_set_tree_vertices function.
+   * \param [in] dim  0 <= \a dim <= 3. Element dimension in the parametric space.
+   * E.g. \a dim = 2 for a \ref T8_ECLASS_QUAD element.
+   */
+  t8_geometry_lagrange (int dim);
+
+  /* Base constructor with no arguments. We need this since it
+   * is called from derived class constructors.
+   * Sets dimension and name to invalid values. */
+  t8_geometry_lagrange (): t8_geometry_with_vertices ()
+  {
+  }
+
+  virtual ~t8_geometry_lagrange ();
+
```

Comment:
```suggestion

  /**
   * Get the type of this geometry.
   * \return The type.
   */
  inline t8_geometry_type_t
  t8_geom_get_type () const
  {
    return T8_GEOMETRY_TYPE_LAGRANGE;
  };

```
We have reworked the `is_geom...` functionality and use an enum now. You can just merge main, remove the old implementation in your code and add the `T8_GEOMETRY_TYPE_LAGRANGE` enum in the https://github.com/DLR-AMR/t8code/blob/main/src/t8_geometry/t8_geometry.h